src/hooks/sync-sites/sync-sites-context.tsx

```diff
@@ -1,24 +1,67 @@
-import React, { createContext, useContext, useState } from 'react';
+import { __, sprintf } from '@wordpress/i18n';
+import React, { createContext, useCallback, useContext, useState } from 'react';
+import { getIpcApi } from '../../lib/get-ipc-api';
 import { SyncSite } from '../use-fetch-wpcom-sites';
-import { usePullPushTimestamps } from '../use-pull-push-timestamps';
+import { useFormatLocalizedTimestamps } from '../use-format-localized-timestamps';
 import { useListenDeepLinkConnection } from './use-listen-deep-link-connection';
 import { useSiteSyncManagement } from './use-site-sync-management';
 import { useSyncPull } from './use-sync-pull';
 import { useSyncPush } from './use-sync-push';
 
 export type SyncSitesContextType = ReturnType< typeof useSyncPull > &
 	ReturnType< typeof useSyncPush > &
-	ReturnType< typeof useSiteSyncManagement > &
-	ReturnType< typeof usePullPushTimestamps >;
+	ReturnType< typeof useSiteSyncManagement > & {
+		getLastSyncTimeText: ( timestamp: string | null, type: 'pull' | 'push' ) => string;
+		updateSiteTimestamp: ( site: SyncSite, type: 'pull' | 'push' ) => Promise< void >;
+	};
 
 const SyncSitesContext = createContext< SyncSitesContextType | undefined >( undefined );
 
 export function SyncSitesProvider( { children }: { children: React.ReactNode } ) {
+	const { formatRelativeTime } = useFormatLocalizedTimestamps();
 	const [ pullStates, setPullStates ] = useState< SyncSitesContextType[ 'pullStates' ] >( {} );
+	const [ connectedSites, setConnectedSites ] = useState< SyncSite[] >( [] );
+
+	const getLastSyncTimeText = useCallback(
+		( timestamp: string | null, type: 'pull' | 'push' ): string => {
+			if ( timestamp === null ) {
```

Comment:
For the case where the timestamp is undefined we were displaying "1 minute ago".
I think this could be solved by checking if timestamp is a nullish value.

```suggestion
		( timestamp: string | null | undefined, type: 'pull' | 'push' ): string => {
			if ( ! timestamp ) {
```

Reply:
Nice catch - fixed in https://github.com/Automattic/studio/pull/714/commits/870df3712c6318609f6f80df01883e68be27bcf7